justfile

```diff
@@ -25,37 +28,41 @@ venv:
 
 # Download data from RNAcentral
 download:
-    curl -O https://ftp.ebi.ac.uk/pub/databases/RNAcentral/r2dt/1.4/cms.tar.gz
+    curl -O https://ftp.ebi.ac.uk/pub/databases/RNAcentral/r2dt/{{ data_version }}/cms.tar.gz
     tar -xzf cms.tar.gz
 
 # Run shell in docker
 run:
-  docker run {{platform}} -v $(pwd):/rna/r2dt -v {{data_dir}}:/rna/r2dt/data/cms -it --rm {{image}}
+    docker run {{ platform_arg }} -v $(pwd):/rna/r2dt -v {{ data_dir }}:/rna/r2dt/data/cms -it --rm {{ image }}
 
 # Run all tests in docker
 test-all:
-  docker run {{platform}} --rm -it -v ./:/rna/r2dt/ -v {{data_dir}}:/rna/r2dt/data/cms {{image}} bash -c "./r2dt.py test"
+    docker run {{ platform_arg }} --rm -it -v ./:/rna/r2dt/ -v {{ data_dir }}:/rna/r2dt/data/cms {{ image }} bash -c "./r2dt.py test"
+
 # Run specific test in docker
 test TEST:
-  docker run {{platform}} --rm -it -v ./:/rna/r2dt/ -v {{data_dir}}:/rna/r2dt/data/cms {{image}} bash -c "./r2dt.py test Test{{TEST}}"
+    docker run {{ platform_arg }} --rm -it -v ./:/rna/r2dt/ -v {{ data_dir }}:/rna/r2dt/data/cms {{ image }} bash -c "./r2dt.py test Test{{ TEST }}"
 
 # Build R2DT Docker image
-build:
-  docker buildx build {{platform}} -t {{image}}  .
+build *BUILD_ARGS:
```

Comment:
Good idea. Added a new command and documented it

Reply:
The new `tag-build` is awesome, thank you! Could it apply the same tag to the resulting R2DT image (currently it becomes `latest` which might be a bit dangerous)?

Reply:
> Could it apply the same tag to the resulting R2DT image (currently it becomes latest which might be a bit dangerous)?

I have two questions here:

1. What do you mean by applying the same tag to the resulting image? `tag-build` already builds the resulting R2DT image…
2. What's dangerous in having the `latest` tag locally by default? Isn't it what every default docker build does? I understand dangers of _pushing_ `latest` to DockerHub, but pushing is done via GH Actions and they don't push `latest` anyway.

I'm a bit confused, as you might notice :)

Reply:
Sure, let me clarify:

> What do you mean by applying the same tag to the resulting image?

If I do `just tag-build pr-xxx` I would like to get `rnacentral/r2dt:pr-xxx` as a result. Why? Read on!

> What's dangerous in having the latest tag locally by default?

Imagine a situation when you review `pr-xxx`, find a bug 😱, report it on GH, and switch to something else. Later on you do `just run`, but because `tag-build` created an image tagged with `latest`, now you are running not a bug-free `latest` image, but an image based on the `pr-xxx` branch. This could create some nasty and hard to find bugs 🐛

Feel free to push back if I am overthinking it 🤔

Reply:
Ok, I see what you mean now. I've pushed the change: now `just tag-build TAG` builds both base and main images with this tag. But now you have to remember `just run` and `just test` work with the `latest` image, not with your custom one :)